### PR TITLE
ACS-5449 Bump commons-net from 3.8.0 to 3.9.0

### PR DESCRIPTION
**Pull request by viepovsky**
After upgrading there were only one test failing, Domenico found out that it was due to change in trust behaviour, FTP client does not longer trust given IP, this change can be reversed by calling `FTPClient#setIpAddressFromPasvResponse(true) `.

Apart from that, the `FTPServerTest` is also fixed with most of the PMD scan warnings removed(just code quality improvements)

### CODE REVIEW

**Comment by viepovsky** [comment]
error

File: repository/src/test/java/org/alfresco/filesys/FTPServerTest.java [hunk 1]
```diff
@@ -216,7 +216,7 @@ public void testFTPConnectNegative() throws Exception
             FTPFile[] files = ftp.listFiles();
             
             assertNotNull(files);
-            assertTrue(files.length == 0);
+            assertEquals(0, files.length);
```
Comment (dsibilio) [inline repository/src/test/java/org/alfresco/filesys/FTPServerTest.java:219]: Let's add a message to this assertion.

File: repository/src/test/java/org/alfresco/filesys/FTPServerTest.java [hunk 2]
```diff
@@ -255,7 +255,7 @@ public void testCWD() throws Exception
             reply = ftp.getReplyCode();
             assertTrue(FTPReply.isPositiveCompletion(reply));
 
-            assertTrue(files.length == 1);
+            assertEquals(1, files.length);
```
Comment (dsibilio) [inline repository/src/test/java/org/alfresco/filesys/FTPServerTest.java:258]: Let's add a message to this assertion.

File: repository/src/test/java/org/alfresco/filesys/FTPServerTest.java [hunk 3]
```diff
@@ -853,23 +853,23 @@ public void testFTPConnectExternalAddressSet() throws Exception
...
+                assertEquals("Client should be in passive mode now", FTPClient.PASSIVE_REMOTE_DATA_CONNECTION_MODE, ftp.getDataConnectionMode());
 
                 reply = ftp.getReplyCode();
                 //see https://www.ietf.org/rfc/rfc959.txt
-                assertTrue("reply code should be 227", reply == 227);
+                assertEquals("reply code should be 227", 227, reply);
 
                 String replyLine = ftp.getReplyString();
-                assertTrue(replyLine != null);
+                assertNotNull(replyLine);
```
Comment (dsibilio) [inline repository/src/test/java/org/alfresco/filesys/FTPServerTest.java:863]: Let's add a message to this assertion.

File: repository/src/test/java/org/alfresco/filesys/FTPServerTest.java [hunk 4]
```diff
@@ -853,23 +853,23 @@ public void testFTPConnectExternalAddressSet() throws Exception
...
                 String encodedImprobableIPAddress = improbableIPAddress.replaceAll("\\.", ",");
                 assertTrue("Pasv command should contain the set external address encoded", replyLine.contains(encodedImprobableIPAddress));
 
                 // now attempt to list the files and check that the command does not succeed
                 FTPFile[] files = ftp.listFiles();
                 assertNotNull(files);
```
Comment (dsibilio) [inline repository/src/test/java/org/alfresco/filesys/FTPServerTest.java:869]: Let's add a message to this assertion.

**Review by dsibilio — COMMENTED** [review]
The fix looks good to me.

Since we have the PMD scan results and the chance to fix some code quality issues on this **test** class, let's do that.

E.g.:
- all `private final` Strings should actually be `private static final` instead;
- `private final Log logger = LogFactory.getLog(...)` should really be `private static final Logger LOGGER = LoggerFactory.getLogger(...)`
- there's a lot of assertions without message
- logging statements should not look like this: `logger.debug("file name=" + file.getName())` but should use parameterized arguments instead, e.g.: `LOGGER.debug("File name = {}", file.geName());`
- there might be more... just look at the PMD report attached to this workflow, for the `FTPServerTest` class

File: repository/src/test/java/org/alfresco/filesys/FTPServerTest.java [hunk 5]
```diff
@@ -765,7 +765,7 @@ public void testModificationTime() throws Exception
             
             // List the files in the new directory
             FTPFile[] files = ftp.listFiles();
-            assertTrue("files not empty", files.length == 0);
+            assertEquals("files not empty", 0, files.length);
```
Comment (dsibilio) [inline repository/src/test/java/org/alfresco/filesys/FTPServerTest.java:768]:
```suggestion
            assertEquals("there should be no files", 0, files.length);
```

File: repository/src/test/java/org/alfresco/filesys/FTPServerTest.java [hunk 6]
```diff
@@ -787,7 +787,7 @@ public void testModificationTime() throws Exception
             
             // Get the new file
             FTPFile[] files2 = ftp.listFiles();
-            assertTrue("files not one", files2.length == 1);
+            assertEquals("files not one", 1, files2.length);
```
Comment (dsibilio) [inline repository/src/test/java/org/alfresco/filesys/FTPServerTest.java:790]:
```suggestion
            assertEquals("there should be only 1 file", 1, files2.length);
```

**Comment by viepovsky** [comment]
I have made requested changes, also deleted one unused test, and changed some boolean variable names.

**Review by dsibilio — APPROVED** [review]

**Review by mikolajbrzezinski — APPROVED** [review]

**Pull request merged**